library/std/src/sys/windows/stdio.rs

```diff
@@ -59,22 +78,73 @@ fn write(handle_id: c::DWORD, data: &[u8]) -> io::Result<usize> {
         return ret;
     }
 
-    // As the console is meant for presenting text, we assume bytes of `data` come from a string
-    // and are encoded as UTF-8, which needs to be encoded as UTF-16.
+    if incomplete_utf8.len > 0 {
+        assert!(
+            incomplete_utf8.len < 4,
+            "Unexpected number of bytes for incomplete UTF-8 codepoint."
+        );
+        if data[0] >> 6 != 0b10 {
+            // not a continuation byte - reject
+            incomplete_utf8.len = 0;
+            return Err(io::Error::new_const(
+                io::ErrorKind::InvalidData,
+                &"Windows stdio in console mode does not support writing non-UTF-8 byte sequences",
```

Comment:
Wouldn't this mean that an incomplete write followed by a complete write will cause the complete write to trigger error, and wouldn't be outputted at all?

Reply:
You mean trying to write invalid UTF-8 to the console? This returns an error right now as well on Windows.

Reply:
Well, if I do an incomplete write followed a complete write, currently the error is reported at the incomplete write; with the change, the error will be reported at the complete write, which might be confusing.